examples/gcoap/gcoap_cli.c

```diff
@@ -35,6 +35,7 @@ static ssize_t _stats_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len, void *c
 static ssize_t _riot_board_handler(coap_pkt_t* pdu, uint8_t *buf, size_t len, void *ctx);
 
 /* CoAP resources */
+/* must be sorted by path (ASCII order) */
```

Comment:
actually its in the documentation: see http://api.riot-os.org/group__net__gcoap.html under *Server Operation* it says: 

> Create an array of resources (coap_resource_t structs) ordered by the resource path, specifically the ASCII encoding of the path characters (digit and capital precede lower case).

I understand that many start by experimenting with and extending of examples, but to me it doesn't make sense to repeat existing present documentation where ever that part is used. Also the documentation is more specific than the suggested comment.

Reply:
@smlng, if you don't object too strongly, I'm OK with this addition. I agree that in general we don't want to repeat ourselves because that can lead to inconsistency. However, this sorting requirement has tripped up people before, which is why we refined the documentation in the header, so maybe repeating ourselves is worthwhile in this case. In addition the nanocoap_server example implementation file already includes exactly the same comment added here, and of course the change doesn't affect the executable.

Reply:
@kb2ma full agree
@smlng I agree that repeated documentation is not good, but it is not intentional that the array of resources must be sorted. There is no hint by naming `_resources[]`. Would you prefer a rename of the array to `_sorted_resources[]` over an extra documentation?

Reply:
I don't strongly object, but then please integrate the comment with other one, i.e. it should fit in one line without getting over 80 chars.